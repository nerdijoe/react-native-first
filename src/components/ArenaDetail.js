import React from 'react'
import {
  View,
  Text,
  Image,
  ScrollView
} from 'react-native'


const styles = {

  container: {
    display: 'flex',
    flex: 1,
    padding: 20,
    marginTop: 40,
    marginBottom: 30,
    flexDirection: 'column',
    flexWrap: 'nowrap',
    alignItems: 'center',


  },
  card: {
    backgroundColor: 'white',
    flexDirection: 'row',
    margin: 5,
    borderRadius: 8,
    borderStyle: 'solid',
    alignItems: 'center'
  },
  title: {
    fontSize: 25,
    fontWeight: 'bold'
  },
  subtitle: {
    fontSize: 10,
    fontWeight: 'bold',
    marginTop: 5
  },
  cardDesc: {
    fontSize: 20,
    padding: 10
  },
  descContainer: {

    borderRadius: 8,
    borderStyle: 'solid',
    borderWidth: 1,
    margin: 10,
    padding: 10,

  },
  cardImage: {
    width: 518/2,
    height: 539/2,
    padding: 10,
    margin: 10,
  },
}

/*
"_id":"58d176be6d804500104b1315",
"idName":"training-camp",
"number":0,
"name":"Training Camp",
"victoryGold":0,
"minTrophies":0,
"order":0,
"__v":0,

*/

class ArenaDetail extends React.Component {
  constructor(props) {
    super(props)
    console.log(this.props)
  }

  render() {
    return (
      <ScrollView >
        <View style={styles.container}>
          <Text style={styles.title}>{this.props.arena.name}</Text>

          <Image
            style={styles.cardImage}
            source={{uri: `http://www.clashapi.xyz/images/arenas/${this.props.arena.idName}.png`}}
          />
          <View style={styles.descContainer}>
            <Text style={styles.subtitle}>Victory Gold</Text>
            <Text>{this.props.arena.victoryGold}</Text>

            <Text style={styles.subtitle}>Minimum Trophies</Text>
            <Text>{this.props.arena.minTrophies}</Text>

          </View>
        </View>
      </ScrollView>
    )
  }
}

export default ArenaDetail
