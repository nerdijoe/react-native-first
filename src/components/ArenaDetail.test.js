import React from 'react'
import renderer from 'react-test-renderer'

import ArenaDetail from './ArenaDetail'

describe('ArenaDetail', () => {
  let arena = {
    "_id":"58d176be6d804500104b1315",
    "idName":"training-camp",
    "number":0,
    "name":"Training Camp",
    "victoryGold":0,
    "minTrophies":0,
    "order":0,
    "__v":0,
    "leagues":[],
    "cardUnlocks":
      ["58d176a56d804500104b1248","58d176a56d804500104b124c","58d176a56d804500104b124d","58d176a56d804500104b124e","58d176a56d804500104b124f","58d176a56d804500104b1252","58d176a56d804500104b1253","58d176a56d804500104b1255","58d176a56d804500104b1257","58d176a56d804500104b1258","58d176a56d804500104b125b","58d176a56d804500104b125d"],
    "chests":[],
    "clan": {"donate":{"common":0,"rare":0},
    "request":  {"common":0,"rare":0}}
  }

  it('renders without crashing', ()=> {
    const tree = renderer.create(<ArenaDetail arena={arena}/>)
    expect(tree).toMatchSnapshot()
  })
})
