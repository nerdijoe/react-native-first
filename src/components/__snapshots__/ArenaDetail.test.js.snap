// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArenaDetail renders without crashing 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "marginBottom": 30,
          "marginTop": 40,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Training Camp
      </Text>
      <Image
        source={
          Object {
            "uri": "http://www.clashapi.xyz/images/arenas/training-camp.png",
          }
        }
        style={
          Object {
            "height": 269.5,
            "margin": 10,
            "padding": 10,
            "width": 259,
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "margin": 10,
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          Victory Gold
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          0
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          Minimum Trophies
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          0
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
